client/my-sites/post-type-list/index.jsx

```diff
@@ -153,6 +154,7 @@ class PostTypeList extends Component {
 						siteId={ siteId }
 						query={ { ...query, page } } />
 				) ) }
+				{ siteId && <QueryPostTypes siteId={ siteId } /> }
```

Comment:
>Regarding the queries, it seems multiple queries should have a way to not run if there's another that was already run by a parent within a certain interval.

It will consolidate if there are multiple components mounting `<QueryPostTypes />` at the same time (or for that matter, if a request is in progress at all), from [`requestingPostTypes` checks in the component](https://github.com/Automattic/wp-calypso/blob/178d0639c1ba80384596ce9f2868ae3ebdde5f89/client/components/data/query-post-types/index.jsx#L22).

>But it still seems that `QueryPostTypes` is something I would run in my `main.jsx` component, not on the list (which would run the `QueryPosts`).

I'd probably disagree with this point in that it moves us back towards our original problems of ancestor components needing to be concerned with the data requirements of their descendants (rather than the components being self-sufficient) and the burden on the developer to remember to do so.